### PR TITLE
document the use of ssh for submodules for submodules_on_private_repo

### PR DESCRIPTION
**Pull request by alexmadon**
see
https://platformsh.zendesk.com/agent/tickets/57833
https://platformsh.atlassian.net/browse/OP-4894
https://platformsh.slack.com/archives/C0JBB9B4G/p1542848286637900
https://support.deploybot.com/article/70-how-do-i-handle-authenticating-to-private-submodules

### CODE REVIEW

File: src/development/submodules.md [hunk 1]
```diff
@@ -43,3 +43,33 @@ E: Error validating submodules in tree:
...
+
+to
+
+```
+[submodule "support/mymodule"]
+    path = support/mymodule
+    url=git@bitbucket.org:myusername/mymodule.git
+```
+
+2. to add a key in the project settings "Deploy Key" tab as per the [Private Repository](development/private-repository.md) documentation page, to allow our git service to pull the module from the remote git service.
```
Comment (Crell) [inline src/development/submodules.md]: It's unclear from this sentence if users add a key to our settings or to the BitBucket settings page.

Comment (alexmadon) [reply src/development/submodules.md]: to our settings

Comment (Crell) [reply src/development/submodules.md]: Please update the text accordingly then so it's less unclear.

File: src/development/submodules.md [hunk 2]
```diff
@@ -43,3 +43,33 @@ E: Error validating submodules in tree:
...
+
+to
+
+```
+[submodule "support/mymodule"]
+    path = support/mymodule
+    url=git@bitbucket.org:myusername/mymodule.git
+```
+
+2. to add the ssh public key in the platform.sh project settings "Deploy Key" tab in our Web UI as per the [Private Repository](development/private-repository.md) documentation page, to allow our git service to pull the module from the remote git service. This assumes of course you have configured the remote git repository to allow this by generating a private/public key pair, see this [bitbucket](https://confluence.atlassian.com/bitbucket/use-ssh-keys-in-bitbucket-pipelines-847452940.html) page for instance.
```
Comment (Crell) [inline src/development/submodules.md]: * "To add", capital T.
* SSH, not ssh.  (Here and above.)
* Git, not git. It's a product name.  (Here and above.)
* Platform.sh.  Capitalize us. :-)
* Bitbucket. Capitalize them.
* "of course" is unnecessary, remove it.
* ", to allow" should be ", which will allow".
* "key pair, see this" is a run-in.  It should be "key pair. For example, see the [Bitbucket documentation]".

**Pull request merged**